packages/react-dom/src/__tests__/ReactDOMServerPartialHydration-test.internal.js

```diff
@@ -3945,4 +3945,94 @@ describe('ReactDOMServerPartialHydration', () => {
       "onRecoverableError: Hydration failed because the server rendered text didn't match the client.",
     ]);
   });
+
+  it('hides a dehydrated suspense boundary if the parent resuspends', async () => {
+    let suspend = false;
+    let resolve;
+    const promise = new Promise(resolvePromise => (resolve = resolvePromise));
+    const ref = React.createRef();
+
+    function Child({text}) {
+      if (suspend) {
+        throw promise;
+      } else {
+        return text;
+      }
+    }
+
+    function Sibling({resuspend}) {
+      if (suspend && resuspend) {
+        throw promise;
+      } else {
+        return null;
+      }
+    }
+
+    function Component({text}) {
+      return (
+        <Suspense>
+          <Child text={text} />
+          <span ref={ref}>World</span>
+        </Suspense>
+      );
+    }
+
+    function App({text, resuspend}) {
+      const memoized = React.useMemo(() => <Component text={text} />, [text]);
```

Comment:
Memoization is important because otherwise the change in props due to the update can force a hydration.